torch/lib/THD/base/data_channels/DataChannelNccl.cpp

```diff
@@ -440,10 +440,15 @@ void DataChannelNccl::allReduce(std::vector<at::Tensor>& data,
                              ncclOp[operation],
                              (*comms)[i],
                              stream));
-    THCudaCheck(cudaEventRecord((*events)[i], stream));
   }
   NCCL_CHECK(ncclGroupEnd());
 
+  for (size_t i = 0; i < data.size(); ++i) {
+    gpuGuard.setDevice(data[i].get_device());
+    auto stream = THCState_getCurrentStream(THDGetCudaState());
```

Comment:
not that it matters a lot, but if you use getCurrentStreamOnDevice here (you already know the device from get_device() call), it will save you a cudaGetDevice call per iteration.

Reply:
Yep, changed to OnDevice